### ISSUE TITLE
Temperature and humidity sensors stopped working

### ISSUE THREAD

**Issue by mircicd** [AnaviTech/anavi-light-controller#3]
Hello

I did not find any official support channels so I'm putting my (most likely user-)issue here: I have been using the Light Controller for a good amount of time now without any problems. I've hooked up a few additional sensors to track temperature, humidity and light intensity: HTU21D, BH1750 and SHT31 (for which I have added support myself). All the sensors have worked flawlessly for more than a year.

I'm running the Light Controller in a Home Assistant Setup. The problem is that all of a sudden I stopped getting sensor data into my MQTT broker, for any of the sensors. I can still send commands to the Light Controller to control the lights, but it just won't send any sensor data. Of course I did several resets and flashed the original firmware but without success.

Is this a known issue or is there any way to debug to find out what the problem might be? I'm using Mosquitto and these are the only relevant logs I could find:
```
Tue Apr 19 18:54:15 2022: New connection from 192.168.1.27 on port 1883.
Tue Apr 19 18:54:15 2022: Client anavi-light-controller-4af09a0ea709ffdc47d2a9fc1cf81f9 has exceeded timeout, disconnecting.
Tue Apr 19 18:54:15 2022: Socket error on client anavi-light-controller-4af09a0ea709ffdc47d2a9fc1cf81f9, disconnecting.
Tue Apr 19 18:54:15 2022: New client connected from 192.168.1.27 as anavi-light-controller-4af09a0ea709ffdc47d2a9fc1cf81f9 (c1, k15, u'').
```

Any help is appreciated!

**Comment by leon-anavi** [comment]
Hi @mircicd,

As an entirely open source project this is an appropriate place to report this type of issues.

Are the sensor modules still properly working as hardware and detected by their I2C addresses?

Have you tried flashing the latest version of original open source firmware?

Would you like to contribute your modifications for SHT31-D as a GitHub pull request to https://github.com/AnaviTechnology/anavi-light-controller-sw ?

Best regards,
Leon

**Comment by mircicd** [comment]
Hi Leon

As per your suggestion I took out the sensors and tested them individually and one by one on a different device. Turned out that the SHT31 sensor was broken! And for some strange reason, leaving it connected to the Light Controller causes all sensor data to stop. After having removed only the broken sensor, I now get sensor readings again as expected. Thanks a lot for your push to the right direction.

I will gladly open a pull request to add SHT31-D support once I find some time to clean up and polish my code.

Best regards,
Dejan
